src/Test/Utilities/DiagnosticAnalyzerTests.Extensions.cs

```diff
@@ -140,10 +140,10 @@ private static string FormatDiagnostics(DiagnosticAnalyzer analyzer, params Diag
                         }
                         else
                         {
-                            Assert.True(location.IsInSource,
+                            Assert.False(location.IsInMetadata,
```

Comment:
There are tests that validate that the diagnostic is fired in the public api file and that's not a source location.